Remove Python 2 build and deps

sopelj opened the pull request.

Commit:
Remove Python 2 build and deps

Commit:
Update syntax

bloom1 commented on debian/control:
```diff
@@ -5,51 +5,17 @@ Maintainer: Wazo Maintainers <dev@wazo.community>
-X-Python-Version: >= 2.7
-X-Python3-Version: >= 3.5
-
-Package: xivo-fetchfw
```
Since you remove this debian package, it is also necessary to mirror the change in the `wazo-platform` meta-package: https://github.com/wazo-platform/wazo-platform/blob/master/debian/control#L72

Also, usually when the debian packaging is modified, it is necessary to add a new changelog in the `debian/changelog` file. You can generate it with `dch -i` and modify the generated log.

sopelj replied:
Oh ok, what do you need to add to the changelog? When I try to generate it, it gives an old date and tag. Looks like it needs to be linked to a release/build too.

sopelj replied:
Actually, now that I look back on the changes, should we keep the meta package xivo-fetchfw? I'm not sure why it was split into xivo-fetchfw and xivo-lib-fetchfw.

bloom1 replied:
I think xivo-fetchfw is not really a metapackage in fact... it is the package that installs the `xivo-fetchfw`executable. So yes we should keep it.

You can edit the `changelog` file manually and put the right information. The release is usually wazo-dev-buster. You can check in any other main service changelog (good place to check is wazo-auth) and see what has been done.

sopelj replied:
That's very strange. I wonder why its setup like that. OK, thanks! I'll put it back and see if I can generate a changelog.

sopelj replied:
Wazo-auth has no entries in the changelog since 2017, I'll try another.

bloom1 commented on setup.py:
```diff
@@ -1,10 +1,7 @@
-from __future__ import absolute_import
-from setuptools import setup
-from setuptools import find_packages
+from setuptools import setup, find_packages
```
imports should be sorted alphabetically.

bloom1 commented on xivo_fetchfw/download.py:
```diff
@@ -1,16 +1,16 @@
 from binascii import b2a_hex
+from urllib import request
+from urllib.error import HTTPError, URLError
+from urllib.request import HTTPPasswordMgrWithDefaultRealm, HTTPBasicAuthHandler, HTTPDigestAuthHandler, ProxyHandler
```
Imports should be sorted alphabetically

bloom1 commented on xivo_fetchfw/package.py:
```diff
@@ -75,10 +71,10 @@ def new_installed_package(self):
 
     def __str__(self):
-        return "%s %s" % (self.pkg_info['id'], self.pkg_info['version'])
+        return "{} {}".format(self.pkg_info['id'], self.pkg_info['version'])
```
Consistency: on line 96 you used the format-string form for the same kind of string.
```python3
return f"{self.pkg_info['id']} {self.pkg_info['version']}"
```

bloom1 commented on xivo_fetchfw/package.py:
```diff
@@ -232,7 +228,7 @@ def _uninstall_pkg(self, installed_pkg, root_dir):
                 removed_paths.append(removed_path)
         except Exception as e:
-            logger.error("Error while removing files of pkg %s: %s" % (pkg_id, e))
+            logger.error(f"Error while removing files of pkg {pkg_id}: {e}")
```
I know that it was already used wrongly in the previous version, but a logger should be lazy for string interpolation. Usually we let the logger library do the job.
In this case the syntax to do so would be:
```python3
logger.error("Error while removing files of pkg %s: %s", pkg_id, e)
```

sopelj replied:
Oh ok, any particular reason? They both only happen if they reach that point in the code. I've seen both everywhere. Just curious.

bloom1 commented on xivo_fetchfw/package.py:
```diff
@@ -596,7 +592,7 @@ def filter_fun(dep_pkg_id):
 
-class UpgraderController(object):
+class UpgraderController:
     # TODO add comments
```
This TODO could have been removed. That is not necessary in this case :sweat_smile:

bloom1 commented on xivo_fetchfw/storage.py:
```diff
@@ -1,17 +1,15 @@
 import json
 import os
 from binascii import a2b_hex
+from configparser import RawConfigParser, ParsingError
```
Import sorting :smile_cat:

bloom1 commented on xivo_fetchfw/tests/test_storage.py:
```diff
@@ -1,14 +1,12 @@
 import tempfile
 import unittest
+from configparser import RawConfigParser
+from unittest.mock import Mock, MagicMock
```
Import sorting :safety_vest:

sopelj replied:
I'm all for import sorting, but if it blocks PRs we need to add isort ASAP, because it's annoying to do manually. 😅

bloom1 reviewed (changes requested):
Great job! Only a few changes necessary :smile:

Commit:
Sort imports, change logger and use f-string as per code review

Commit:
Restore meta/installer package and update changelog

bloom1 committed:
debian: add datetime to 23.01 changelog

Why: 23.01 is considered a higher version than any version with a ~
followed by something else. This means that any development version
installed on a machine will keep the development version in case the
package is built using only the 23.01 version number.

bloom1 reviewed (approved):
Great work :smile:
